### ISSUE TITLE
Some problems while pretraining in multiflow dataset 

### ISSUE THREAD

**Issue by Lingoererer** [uzh-rpg/deep_ev_tracker#10]
Thank you for your great contribution! But I followed your tutorial and didn't achieve the same effect, The details are as follows:

1. I downloaded the multiflow dataset and generate ground-truth tracks and input event representations with dt=0.0200 , representation = "time_surfaces_v2_5"  both on train and test split.
![image](https://github.com/uzh-rpg/deep_ev_tracker/assets/59367063/4502a6a4-99dd-42af-954e-3ccb7c1854d1)

3. On the synthetic data, I kept all the parameters the same as you set and trained the model “correlation3_unscaled” with a RTX 3090, epoch=142 and step=133991. The **details** of the training process are as follows:

**the file "train.log" is as follows:**
`[2023-08-26 18:47:00,640][__main__][INFO] - 
data:
  name: mf
  _target_: utils.dataset.MFDataModule
  data_dir: /zssd/monsoon2/event_camera/dataset
  extra_dir: /zssd/monsoon2/event_camera/dataset
  dt: 0.02
  num_workers: 4
  batch_size: 32
  n_train: 30000
  n_val: 1000
  augment: true
  global_mode: false
  mixed_dt: true
  representation: time_surfaces_v2_5
  track_name: shitomasi_custom
  patch_size: 31
model:
  optimizer:
    _target_: torch.optim.Adam
    lr: 0.0001
  name: correlation3_unscaled
  _target_: models.correlation3_unscaled.TrackerNetC
  patch_size: 31
  feature_dim: 384
  representation: time_surfaces_v2_5
  n_vis: 2
  init_unrolls: 4
  max_unrolls: 23
  debug: false
  pose_mode: false
checkpoint_path: none
init_unrolls: 4
max_unrolls: 23
unroll_factor: 4
unroll_schedule:
- 80000
- 120000
- 140000
experiment: open_source
track_name: shitomasi_custom
representation: time_surfaces_v2_5
patch_size: 31
debug: false
n_vis: 2
logging: true
trainer:
  benchmark: true
  log_every_n_steps: 10
  max_epochs: 1000
  num_processes: 1
  num_sanity_val_steps: 1` 

**the loss pictures:**
![image](https://github.com/uzh-rpg/deep_ev_tracker/assets/59367063/c3979b7f-c420-46ac-b43f-055f39b2d711)
**the visual result:**
![image](https://github.com/uzh-rpg/deep_ev_tracker/assets/59367063/a69bae64-d676-4645-9d31-8ec81cdea356)
![image](https://github.com/uzh-rpg/deep_ev_tracker/assets/59367063/0b117ddd-5594-414c-ba5e-8ef49feb28c8)

4. I evaluated both on real datasets **EDS** and **EC** using my trained weights and the mf_pretrained weights you provided. I used the code you provided to calculate the benchmark metrics, but the gap is very large, especially in the two metrics of feature age and expected feature age. The detailed comparison is as follows:(**left** : my trained ckpt  **right**: your pretrained ckpt)

![image](https://github.com/uzh-rpg/deep_ev_tracker/assets/59367063/e5d572df-b11d-4048-ad3d-78f684f9f72f)
![image](https://github.com/uzh-rpg/deep_ev_tracker/assets/59367063/eaf31c2f-05af-4e1b-8c58-3c656e9a3b9b)

**Comment by MessikommerNico** [comment]
Many thanks for your interest in our work!

Based on the shared "train.log" file, I can not see any difference compared to the training of the model weights we provide in the repo. One thing that changed is the version of the Multiflow dataset, which should not have a big impact on the performance. I would recommend the following steps to find the error:

- Check that all sequences of the Multiflow dataset are considered during training and if the event representations look reasonable.
- The benchmark output looks a bit off. There are identical numbers between the "age_mu" and "expected age" metrics for the EC dataset. Additionally, the values achieved with the "mf_pretrained" are too high. Maybe have there a closer look. 

Let me know if you find a potential error source. I will then also check myself.

**Comment by youthfrog** [comment]
Hi @Lingoererer , I’m currently encountering a similar issue and noticed you mentioned this problem earlier. I saw that the issue has been closed, but there’s no clear indication whether it was resolved on your end.

May I ask — did you manage to solve this problem eventually? If so, could you please share what worked for you (e.g., changes in `dt`, `representation`, or any other settings)? I would really appreciate your help. Thanks in advance!
